### DISCUSSION TITLE
[Feedback]: AK's Test of Team Plan and Bookings

### DISCUSSION THREAD

**Discussion by arthurkallinen** [Shelf-nu/shelf.nu#778] [Share your feedback]
### **Bookings**
- Ability to give custody of a booking to a NVR
- Add "Note" field for extra info
- Add "Person/Company/Client"; Same "client" might have multiple bookings for different events that would be the booking name
- Add Booking ID that is just an automatic up-rolling number for easier recognizing of bookings and amount of them.
- Assume time of booking to "08:00 until 16:00 / 00:00 until 00:00" if else not given.
- Move time before date in the slot, as it is now easy to go over. Also, would be nice to have visual selection box for it like date has the calendar mode.
- Ability to duplicate booking (into draft)

### **Team Plan**
- Ability to cancel invite for an user. Now you have to wait for them the accept and then Cancel
- Ability to change user from Self-service to Admin and vice-versa.
- Ability to toggle "Available for booking" of all items in a certain Location.

### **Custom Fields / Importing .csv**
- When importing Custom Field Type: Option, it creates all rows as new options, even tho these should be parsed together, as mutliple items need to be able to have same option, but duplicates are not good.
Viewed:
.csv: 
![MultipleCSV](https://github.com/Shelf-nu/shelf.nu/assets/44576346/5bbb48bf-fb46-42c6-8462-1cf79ff13b08)
Shelf item view: 
![MultipleShelfs](https://github.com/Shelf-nu/shelf.nu/assets/44576346/2f7ce89a-bc73-43af-8677-6387142e47d6)
Shelf Custom item edit view:
![MultipleRowsDropdown](https://github.com/Shelf-nu/shelf.nu/assets/44576346/2b0172fc-2fec-4e66-a492-398621ac3ce5)
 - ^ if this is to be done inside custom field / option engine; could be done when adding ability to re-order custom fields (earlier feature request of mine, https://github.com/Shelf-nu/shelf.nu/issues/649 )

### Describe the solution you'd like

.

**Comment by DonKoko** [comment]
hey @arthurkallinen . Thanks a lot for all the feedback. It is extremely useful. I will address all your points/questions below.

> Ability to give custody of a booking to a NVR

We are considering this, however there is a big issue with that. The NRM will not be able to get any email notifications as of now.

> Add "Note" field for extra info

Great idea. Thanks for the suggestion.

> Add "Person/Company/Client"; Same "client" might have multiple bookings for different events that would be the booking name

Do you mean like a text field on the booking itself.

> Add Booking ID that is just an automatic up-rolling number for easier recognizing of bookings and amount of them.

We have this planned, however it didn't make the cut in the beta/mvp of bookings. Will be added.

> Assume time of booking to "08:00 until 16:00 / 00:00 until 00:00" if else not given.

I guess we can do that. So if I understand correctly you mean that 00:00 should be the default value?

> Move time before date in the slot, as it is now easy to go over. Also, would be nice to have visual selection box for it like date has the calendar mode.

There are some limitations here. We are all about web standards and we try to keep things as clean as possible, so if we can rely on those and not implement some bloated library for managing date/time selectors we prefer that. So for now we have decided to rely on native browser implementation for date/time selecting. So the way it works/looks basically depends on the browser you are using. We have this on the top of our mind and we are considering it, however for now we would rather stick to browser defaults.

> Ability to duplicate booking (into draft)

Thanks for the suggestion. Would you expect that to duplicate also the date/time and all the assets? 


> Ability to cancel invite for an user. Now you have to wait for them the accept and then Cancel

This is in progress already.

> Ability to change user from Self-service to Admin and vice-versa.

We have considered this but there are quite some complexities related to this so it's not such a straightforward change. We will think further into how it can be done and let you know.

> Ability to toggle "Available for booking" of all items in a certain Location.

Thanks for the feature suggestion. @jurrejansen do you think we can make a design for this? Please also consider edge cases. Like what if an asset in that location is already booked or in custody? How do we handle all those cases.


> When importing Custom Field Type: Option, it creates all rows as new options, even tho these should be parsed together, as mutliple items need to be able to have same option, but duplicates are not good.

I will check that out. Please send the example file I talked about earlier and we can handle this.

**Comment by DonKoko** [comment]
> Ability to cancel invite for a user. Now you have to wait for them the accept and then Cancel

@arthurkallinen this is done and will be live shortly.

**Comment by DonKoko** [comment]
Hey @arthurkallinen . I have transformed your issue into a discussion as it includes many different features inside it. We can keep an overview here and we will create issues for the different points.
